### PR TITLE
oracle-instantclient: support aarch64-darwin

### PR DESCRIPTION
**Pull request by lvitaly**
## Description

Add support for aarch64-darwin (m1) for oracle-instantclient.
https://www.oracle.com/database/technologies/instant-client/macos-arm64-downloads.html

This is a fresh update/duplicate of https://github.com/NixOS/nixpkgs/pull/335529.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [x] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by lvitaly** [comment]
@aca @SuperSandro2000 @joshuata
Hey guys, I hope you are well and okay with duplicating @aca's PR to merge it ASAP 🙏

File: pkgs/by-name/or/oracle-instantclient/package.nix [hunk 1]
```diff
@@ -67,18 +70,33 @@ let
         tools = "sha256-1xFFGZapFq9ogGQ6ePSv4PrXl5qOAgRZWAp4mJ5uxdU=";
         odbc = "sha256-S6+5P4daK/+nXwoHmOkj4DIkHtwdzO5GOkCCI612bRY=";
       };
+      aarch64-darwin = {
+        basic = "1znpm1xkkq7c52nz3vx451q4jn7x6a5dp7jp5lihrxvh71cdpk8v";
+        sdk = "1fycrzc0l1ayprs7qg344axqarm6swz23zdij70bkdly137dzsix";
+        sqlplus = "0xnnlzhv0vdih2a0ycx2pd14rx2y73x2fc31rpmvs09il6cs64wj";
+        tools = "0vp5rpglkscwaaq0w9nbki1icjmpfcq9ws979rv8v9fp0mp943w0";
+        odbc = "1pfqh0r7gr02zw3icrnf70sim66fasy10my7yps1w976grs14fi7";
```
Comment (SuperSandro2000) [inline pkgs/by-name/or/oracle-instantclient/package.nix:78]: Please use SRI hashes

Comment (lvitaly) [reply pkgs/by-name/or/oracle-instantclient/package.nix]: Thanks. Fixed.

**Comment by lvitaly** [comment]
Hey, @SuperSandro2000, I’ve addressed your comment. Please take a look 🙏 Thanks.

**Comment by lvitaly** [comment]
Kind reminder 😇

**Pull request merged**